internal/services/storagecache/managed_lustre_file_system_auto_export_job_resource.go

```diff
@@ -0,0 +1,233 @@
+package storagecache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/storagecache/2024-07-01/autoexportjob"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/storagecache/2024-07-01/autoexportjobs"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/storagecache/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ManagedLustreFileSystemAutoExportJobModel struct {
+	Name               string            `tfschema:"name"`
+	ResourceGroupName  string            `tfschema:"resource_group_name"`
+	AmlFileSystemName  string            `tfschema:"aml_file_system_name"`
+	Location           string            `tfschema:"location"`
+	AutoExportPrefixes []string          `tfschema:"auto_export_prefixes"`
+	AdminStatus        string            `tfschema:"admin_status"`
+	Tags               map[string]string `tfschema:"tags"`
+}
+
+type ManagedLustreFileSystemAutoExportJobResource struct{}
+
+var _ sdk.ResourceWithUpdate = ManagedLustreFileSystemAutoExportJobResource{}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) ModelObject() interface{} {
+	return &ManagedLustreFileSystemAutoExportJobModel{}
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) ResourceType() string {
+	return "azurerm_managed_lustre_file_system_auto_export_job"
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"aml_file_system_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ManagedLustreFileSystemName,
+		},
+
+		"auto_export_prefixes": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
```

Comment:
Number of maximum allowed paths for now is 1, so we should enforce that in the schema.  Since this is an array of `blob paths/prefixes` which defaults to `/` can we also add a better validation function here as well?

```suggestion
		"auto_export_prefixes": {
			Type:     pluginsdk.TypeList,
			Required: true,
			MaxItems: 1,
			Elem: &pluginsdk.Schema{
				Type:         pluginsdk.TypeString,
				ValidateFunc: validation.StringIsNotEmpty,
			},
		},
```

Reply:
Will change.